specification/appendix-d-observability.md

```diff
@@ -0,0 +1,47 @@
+---
+id: appendix-d
+title: "Appendix D: Observability"
+description: Conventions for OpenFeature telemetry signals 
+sidebar_position: 5
+---
+
+# Appendix D: Observability
+
+This document describes conventions for extracting data from the OpenFeature SDK for use in telemetry signals.
+It primarily focuses on providing recommendations for mapping well-known fields in OpenFeature to [OpenTelemetry feature-flag log records](https://opentelemetry.io/docs/specs/semconv/feature-flags/feature-flags-logs/) and other semantic conventions.
+
+## Evaluations
+
+Flag evaluation telemetry comprises data resolved from the provider resolution (evaluation details and flag metadata) as well as metadata about the provider itself.
+This is particularly relevant to telemetry-related [hooks](./sections/04-hooks.md).
+
+### Evaluation Details
+
+The following describes how fields on the [evaluation details](types.md#evaluation-details) are mapped to feature flag log records:
+
+| Evaluation Details Field | Log Record Attribute                    | Notes                                                |
+| ------------------------ | --------------------------------------- | ---------------------------------------------------- |
+| `flag key`               | `feature_flag.key`                      | See: [flag key](./glossary.md#flag-key)              |
+| `error code`             | `error.type`                            | See: [error code](./types.md#error-code)             |
+| `variant`                | `feature_flag.variant`                  | See: [variant](./glossary.md#variant)                |
```

Comment:
There's questions about how exactly to represent `value` (important if provider cannot supply `variant`):

- is it represented in the body?
- can it be of `any` type?

Reply:
> is it represented in the body?

Do you mean the body of the log record?

The [spec](https://opentelemetry.io/docs/specs/semconv/feature-flags/feature-flags-logs/) says: 

```
value: If and only if feature flag provider does not supply variant or equivalent concept. 
Otherwise, value should be treated as opt-in.
```

And the type is `undefined` there as there is no `any` type in OTEL. 
Recently there was also opened a PR to add the `value_type` to the event: https://github.com/open-telemetry/semantic-conventions/pull/1689